### PR TITLE
BUGFIX check for different spelling of session outdated exceptions on…

### LINKED ISSUE #3151: User token is not refreshed with autoRefreshTokens when invalidated by Magento backend
## Current behavior

When user token is invalidated  by Magento backend, refresh logic is not working in /core/lib/sync/task.ts

The error returned by Magento is: 
{"code":500,"result":"The consumer isn't authorized to access self."}

The check if token is invalidated looks like:
resultString.indexOf('not authorized')) >= 0

So now it is not passed as 'not authorized' !== ' isn't authorized'

## Expected behavior

If user token is invalid and autoRefreshTokens === true, user token should be refreshed 

## Steps to reproduce the issue

1) Create user account
2) Wait timeout to user token invalidated on Magento Backend
3) Reload application home page

## Can you handle fixing this bug by yourself?

- [+] YES
- [ ] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer.
Pick one option.

- [ ] This is a bug report for test version on https://test.storefrontcloud.io - In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [ ] This is a bug report for current Release Candidate version on https://next.storefrontcloud.io - In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [ +] This is a bug report for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version hotfix - In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

## Environment details

- Browser:  Chrome Version 75.0.3770.100
- OS:  MacOS 10.13.6
- Node:  11.3.0
- Code Version:  master

### LINKED ISSUE #3178: Session gets lost
## Current behavior

If you are not active in the shop for some time, the session seems to lose its validity although "Remember me" was selected.

An error message (Consumer isn't authorized to access self) will appear in the frontend as "You are not logged in".

But the user is still access the account and navigate to data which is cached, but all new request to the API which requires authentication are throwing errors.

## Expected behavior

The session is not lost, especially if you user selects "remember me".

## Can you handle fixing this bug by yourself?

We already tried to debug it, but currently we have no idea what could be the reason or where to start.

- [ ] YES
- [X] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer.
Pick one option.

- [ ] This is a bug report for test version on https://test.storefrontcloud.io - In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [ ] This is a bug report for current Release Candidate version on https://next.storefrontcloud.io - In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [X] This is a bug report for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version hotfix - In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

## Environment details

- Browser: Chrome
- OS: Mac, Linux
- Node: 8
- Code Version: 1.9.2

### PR DESCRIPTION
**Pull request by danielmaier42**
… magento-side

### Related issues
closes #3178 
closes #3151 

### Short description and why it's useful
As mentioned by nikblack on Issue #3151 and @pkarw on #3178 the token auto refreshing cannot work when the wrong string is checked. Therefore i am submitting this pull request which compares the message not just with "not authorized" (backward compatibility) but also against "isn't authorized".

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [x] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [x] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [x] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

@pkarw - could you please check? 🙂

### CODE REVIEW

File: core/lib/sync/task.ts [hunk 1]
```diff
@@ -65,7 +65,12 @@ function _internalExecute (resolve, reject, task: Task, currentToken, currentCar
     if (jsonResponse) {
       if (parseInt(jsonResponse.code) !== 200) {
         let resultString = jsonResponse.result ? toString(jsonResponse.result) : null
-        if (resultString && (resultString.indexOf(i18n.t('not authorized')) >= 0 || resultString.indexOf('not authorized')) >= 0 && currentToken !== null) { // the token is no longer valid, try to invalidate it
+        if (resultString && currentToken !== null && (
+          resultString.indexOf(i18n.t('not authorized')) >= 0 ||
+          resultString.indexOf('not authorized') >= 0 ||
+          resultString.indexOf(i18n.t('isn\'t authorized')) >= 0 ||
+          resultString.indexOf('isn\'t authorized') >= 0)) {
+          // the token is no longer valid, try to invalidate it
```
Comment (pkarw) [inline core/lib/sync/task.ts]: Please just change it to the check of the single `authorized` word + please add the `authorized` to at least `en_US.csv`

**Comment by patzick** [comment]
@danielmaier42 isn't response code 401 then? Do we need to parse response message?

**Comment by pkarw** [comment]
It seems like Magento2 is setting the right response code:
http://demo-magento2.vuestorefront.io/rest/V1/orders?searchCriteria%5Bfilter_groups%5D%5B0%5D%5Bfilters%5D%5B0%5D%5Bfield%5D=status&searchCriteria%5Bfilter_groups%5D%5B0%5D%5Bfilters%5D%5B0%5D%5Bvalue%5D=pending

Please feel free to adjust the way vue-storefront-api is handling these codes (for reference: https://github.com/DivanteLtd/vue-storefront-api/blob/bb3b91e49a3311a772a437b5177d49a6998ef5f7/src/api/user.js#L163)

**Comment by pkarw** [comment]
The problem is that `magento2-rest-client` library is just ignoring the code - https://github.com/DivanteLtd/magento2-rest-client/blob/ac767431a2f7ca864217f127e33bff3e07c26885/lib/rest_client.js#L47

We should adjust this error handling feature to raise the proper error code along with the message (for example raising catch of Error object that has proper `toString()` and `.code` + check/change all the api calls across `vue-storefront-api`)

**Comment by pkarw** [comment]
Related: https://github.com/DivanteLtd/vue-storefront-api/issues/297

**Comment by pkarw** [comment]
Closing this one in favor of #3237

**Pull request closed without merging**